### PR TITLE
CTCTRADERS-2849: Fixed CYA for option and mandatory previous admin re…

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**
…ferences

### CODE REVIEW

File: app/navigation/annotations/addItemsNavigators/AddItemsAdminReferenceNavigator.scala [hunk 1]
```diff
@@ -125,10 +126,20 @@ class AddItemsAdminReferenceNavigator @Inject() () extends Navigator {
...
     }
 
-  private def removePreviousAdministrativeReference(itemIndex: Index, mode: Mode)(ua: UserAnswers) =
-    ua.get(DeriveNumberOfPreviousAdministrativeReferences(itemIndex)) match {
-      case None | Some(0) => previousReferencesRoutes.AddAdministrativeReferenceController.onPageLoad(ua.lrn, itemIndex, mode)
-      case _              => previousReferencesRoutes.AddAnotherPreviousAdministrativeReferenceController.onPageLoad(ua.lrn, itemIndex, mode)
+  private def removePreviousAdministrativeReference(itemIndex: Index, mode: Mode)(ua: UserAnswers): Call = {
+
+    val isAllowedDeclarationType: Boolean = ua.get(DeclarationTypePage).fold(false)(t2Options.contains)
+    val countryOfDeparture: Boolean       = ua.get(IsNonEuOfficePage).getOrElse(false)
```
Comment (jackheslop96) [inline app/navigation/annotations/addItemsNavigators/AddItemsAdminReferenceNavigator.scala:132]: Very minor but could this val be better named?

File: app/viewModels/AddItemsCheckYourAnswersViewModel.scala [hunk 2]
```diff
@@ -141,11 +141,20 @@ object AddItemsCheckYourAnswersViewModel {
...
       }
 
-    Section(
-      msg"addItems.checkYourAnswersLabel.references",
-      Seq(checkYourAnswersHelper.addAdministrativeReference(index).toSeq, referencesRows).flatten,
-      checkYourAnswersHelper.addAnotherPreviousReferences(index, msg"addItems.checkYourAnswersLabel.references.addRemove")
-    )
+    val addAdministrativeReferenceRow = checkYourAnswersHelper.addAdministrativeReference(index).toSeq
+
+    val rows = addAdministrativeReferenceRow ++ referencesRows
```
Comment (jackheslop96) [inline app/viewModels/AddItemsCheckYourAnswersViewModel.scala:146]: Could this just be `addAdministrativeReferenceRow +: referencesRows` without the `.toSeq` ?

Comment (MatthewAlexanderBeck) [reply app/viewModels/AddItemsCheckYourAnswersViewModel.scala:146]: The AddAdministrativeReferenceRow is an option of SummaryList.Row. The toSeq method unwraps it to either a Seq or a Seq.empty. I think the +: will give a Product in this case :D

Comment (jackheslop96) [reply app/viewModels/AddItemsCheckYourAnswersViewModel.scala:146]: Ah makes sense

**Review by jackheslop96 — APPROVED** [review]

**Review by vajarova — APPROVED** [review]

**Pull request merged**